### ISSUE TITLE
[new-parser] Parsing fails if a Java keyword appears in a qualified name

### ISSUE THREAD

**Issue by yurloc** [apache/incubator-kie#5818]
## Parent issue
- #5678 

## Failing tests
- `org.kie.declarativetypes.JavaBeansEventRoleTest#testImportBean`
- `org.drools.mvel.integrationtests.ExtendsTest#testDeclareExtendsWithFullyQualifiedName`

## Description
- `package` translates to the `PACKAGE` token, which is unexpected in the package's qualified name.
- The rest of the input is ignored and the parser **does not report an error**!
  - This is no longer true thanks to #5930. The parser now reports an error as expected.

## Rule code snippet
```
package some.other.package;
import org.drools.compiler.Bean;
declare Bean
  @role(event)
end
```

## Error output
Updated:
```
### parse : ANTLR4_PARSER_ENABLED = true
line 1:18 mismatched input '.' expecting {<EOF>, 'unit', 'function', 'global', 'declare', 'rule', 'query', 'salience', 'enabled', 'no-loop', 'auto-focus', 'lock-on-active', 'refract', 'direct', 'agenda-group', 'activation-group', 'ruleflow-group', 'date-effective', 'date-expires', 'dialect', 'calendars', 'timer', 'duration', 'import'}
14:07:57.000 [main] ERROR o.d.c.k.b.impl.AbstractKieProject.buildKnowledgePackages:280 - Unable to build KieBaseModel:defaultKieBase
[1,18]: mismatched input '.' expecting {<EOF>, 'unit', 'function', 'global', 'declare', 'rule', 'query', 'salience', 'enabled', 'no-loop', 'auto-focus', 'lock-on-active', 'refract', 'direct', 'agenda-group', 'activation-group', 'ruleflow-group', 'date-effective', 'date-expires', 'dialect', 'calendars', 'timer', 'duration', 'import'}
[0,0]: Parser returned a null Package


org.kie.api.builder.CompilationErrorsException: Unable to create KieModule, Errors Existed: [Message [id=1, kieBase=defaultKieBase, level=ERROR, path=bean1.drl, line=1, column=0
   text=mismatched input '.' expecting {<EOF>, 'unit', 'function', 'global', 'declare', 'rule', 'query', 'salience', 'enabled', 'no-loop', 'auto-focus', 'lock-on-active', 'refract', 'direct', 'agenda-group', 'activation-group', 'ruleflow-group', 'date-effective', 'date-expires', 'dialect', 'calendars', 'timer', 'duration', 'import'}], Message [id=2, kieBase=defaultKieBase, level=ERROR, path=bean1.drl, line=0, column=0
   text=Parser returned a null Package]]

	at org.drools.compiler.kie.builder.impl.KieBuilderImpl.getKieModule(KieBuilderImpl.java:526)
	at org.drools.compiler.kie.builder.impl.KieBuilderImpl.getKieModule(KieBuilderImpl.java:532)
	at org.drools.compiler.kie.builder.impl.KieBuilderImpl.getKieModule(KieBuilderImpl.java:507)
	at org.kie.declarativetypes.JavaBeansEventRoleTest.testImportBean(JavaBeansEventRoleTest.java:108)
```
Originally when this issue was reported:
```
00:14:32.657 [main] WARN  o.d.c.k.builder.impl.KieBuilderImpl.packageNameForFile:396 - File 'bean1.drl' is in folder '' but declares package 'some.other.package'. It is advised to have a correspondance between package and folder names.
### parse : ANTLR4_PARSER_ENABLED = true

org.junit.ComparisonFailure: 
Expecting value to be true but was false 
Expected :true
Actual   :false
<Click to see difference>


	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:67)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at org.kie.declarativetypes.JavaBeansEventRoleTest.testImportBean(JavaBeansEventRoleTest.java:115)
```

**Comment by tkobayas** [comment]
@yurloc  As we discussed in https://github.com/apache/incubator-kie-drools/pull/5819 , we would go for the direction "Add Java keywords to drlIdentifier " for backward compatibility (= focus on passing existing tests).

After the phase, we would re-consider such issues like dropping useless syntax / ambiguity. Feel free to share if you have any concern. Thanks!

**Comment by tkobayas** [comment]
`/take`

**Comment by yurloc** [comment]
> Feel free to share if you have any concern. Thanks!

No concerns. I think this is the best solution for the time being.
- It's 100% backward compatible.
- It fixes the problem. Once we have a green test suite, we can look for an alternative solution.

**Comment by yurloc** [comment]
Fixed by #5958.
